policies/step_annotations.rego

```diff
@@ -0,0 +1,25 @@
+package hacbs.contract.step_annotations
+
+import data.lib
+
+deny[{"msg": msg}] {
+	att := data.attestations[_]
+
+	some step_index
+
+	# null annotations show up when there are none defined
+	# can there be null annotations?
+	att.predicate.buildConfig.steps[step_index].annotations != null
+
+	# if an annotations is defined, it shows up in a list
+	annotations := [x | x := att.predicate.buildConfig.steps[step_index].annotations[_]; not valid_annotation(x)]
+	annotations != []
+	msg := sprintf(
+		"step %d has invalid annotations(s): %s",
+		[step_index, annotations],
+	)
+}
+
+valid_annotation(annotation) {
+	lib.config.allowed_annotations[_] = annotation
```

Comment:
I see that this works, but why is it `=` instead of `==` ?

Reply:
Because rego..?

(I was trying to explain it, but I realized I don't fully get it myself. Do the tests pass with == ?

Reply:
Because rego is all I got too. I originally had ==, but had some issues and tried the single = and never changed it back.

Reply:
I think I get it:
> [Unification](https://www.openpolicyagent.org/docs/latest/policy-language/#unification-)  (=) combines assignment and comparison. Rego will assign variables to values that make the comparison true. Unification lets you ask for values for variables that make an expression true.

Because in this case, the variable is already defined on both sides, this is just a comparison. Replacing it with `==` should have the same result.

> [Best practice](https://www.openpolicyagent.org/docs/latest/policy-language/#best-practices-for-equality) is to use assignment := and comparison == wherever possible. The additional compiler checks help avoid errors when writing policy, and the additional syntax helps make the intent clearer when reading policy.

> Under the hood := and == are syntactic sugar for =, local variable creation, and additional compiler checks.

And to prove this, try this version of the function (it works the same way!):
```rego
valid_annotation(annotation) {
	lib.config.allowed_annotations[_] = spam
	spam == annotation
}
```

I think we should go through the code base and change to `==` whenever possible. Wdyt?